internal/pkg/config/config.go

```diff
@@ -123,12 +154,23 @@ func MustNewConfigFrom(from interface{}) *Config {
 	return c
 }
 
-func newConfigFrom(in *ucfg.Config) *Config {
-	return (*Config)(in)
+func newConfigFrom(in *ucfg.Config, otel *confmap.Conf) *Config {
+	return &Config{
+		Agent: in,
+		OTel:  otel,
+	}
+}
+
+// Unpack implements the ucfg.Unpacker interface.
+func (c *Config) Unpack(val interface{}) error {
+	if c.Agent == nil {
+		c.Agent = ucfg.New()
+	}
+	return c.Agent.Merge(val)
 }
 
-// Unpack unpacks a struct to Config.
-func (c *Config) Unpack(to interface{}, opts ...interface{}) error {
+// UnpackTo unpacks this config into to with the given options.
+func (c *Config) UnpackTo(to interface{}, opts ...interface{}) error {
```

Comment:
as said PR would be o simpler without this one, if we could extract this to refactoring PR we could get reviews for this one faster

Reply:
Yeah, I probably should have split it out. Being that you already gave a +1 I would prefer to leave it as it. Splitting that out is going to take more work, then just getting this merged.